app/views/discovered_hosts/index.html.erb

```diff
@@ -1,3 +1,3 @@
 <% title_actions multiple_discovered_hosts_actions_select -%>
-<% title_actions display_link_if_authorized(_("Auto Provision All"), hash_for_auto_provision_all_discovered_hosts_path, :method => :post) %>
+<% title_actions display_link_if_authorized(_("Auto Provision All"), hash_for_auto_provision_all_discovered_hosts_path, {:method => :post, :disabled => @hosts.empty?}) %>
```

Comment:
looks good, I would consider adding another check in the controller (e.g. when you click on auto provision) that there are no 0 hosts. (I assume the same is relevant for the API once it exists)